### PR TITLE
feat: add format options for report output

### PR DESCRIPTION
**Pull request by rheimus**
## Description

Adds a flag for setting what type of report output is desired
- plain - raw text essentially without colors/symbols
- pretty - (default) same as before with colors symbols

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Related Issue

#133 

## Motivation and Context

Sadly many environments and parsing tools don't like fancy characters or color sequences, this just gives an alternative friendly output for more basic systems.

## How Has This Been Tested?

- Tested manually under WSL windows
- Unit tests added for error output

I could add tests for the 'success' (no errors) path if desired.

## Checklist:

- [x] I have updated the documentation (if required).
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I added a picture of a cute animal cause it's fun
🦍 <-- cute animal

### CODE REVIEW

**Comment by rheimus** [comment]
Interesting note, this repo does not work under native windows, has problems spawning processes for the unit tests.

**Comment by lirantal** [comment]
We'll look into it shortly @rheimus 🙏🏼

File: packages/lockfile-lint/__tests__/cli.test.js [hunk 1]
```diff
@@ -149,6 +149,82 @@ describe('CLI tests', () => {
     })
   })
 
+  test('Unspecified output has symbols and color', done => {
```
Comment (lirantal) [inline packages/lockfile-lint/__tests__/cli.test.js:152]:
```suggestion
  test('default usage has output with symbols and color', done => {
```

File: packages/lockfile-lint/__tests__/cli.test.js [hunk 2]
```diff
@@ -149,6 +149,82 @@ describe('CLI tests', () => {
...
+    })
+
+    process.stderr.on('close', _ => {
+      expect(output).toMatch(/[×✖]/)
+      expect(output).toContain('\x1b[0m')
+      done()
+    })
+  })
+
+  test('Format pretty has symbols and color', done => {
```
Comment (lirantal) [inline packages/lockfile-lint/__tests__/cli.test.js:176]:
```suggestion
  test('When using --pretty formatting then output has symbols and color', done => {
```

File: packages/lockfile-lint/__tests__/cli.test.js [hunk 3]
```diff
@@ -149,6 +149,82 @@ describe('CLI tests', () => {
...
+    })
+
+    process.stderr.on('close', _ => {
+      expect(output).toMatch(/[×✖]/)
+      expect(output).toContain('\x1b[0m')
+      done()
+    })
+  })
+
+  test('Format plain does not have symbols or color', done => {
```
Comment (lirantal) [inline packages/lockfile-lint/__tests__/cli.test.js:202]:
```suggestion
  test('hen using --plain formatting then output does not have symbols or color', done => {
```

**Review by lirantal — CHANGES_REQUESTED** [review]

**Comment by lirantal** [comment]
@rheimus looks good to me to land. Thank you for refactoring to support the error/warn/success functions and building this plain/pretty capability into the lockfile-lint CLI. Much appreciated 👏🙏🏼

**Comment by rheimus** [comment]
Applied the requested changes and fixed a typo in the plain formatting test set.

**Review by lirantal — APPROVED** [review]

**Comment by lirantal** [comment]
Looks great, thank you!

**Pull request merged**